source/differentiation/sd/symengine_optimizer.cc

```diff
@@ -63,10 +63,28 @@ namespace Differentiation
       , dependent_variables_output(0)
       , map_dep_expr_vec_entry(other.map_dep_expr_vec_entry)
       , ready_for_value_extraction(false)
+      , has_been_serialized(false)
```

Comment:
this is unrelated, right?

Reply:
Yeah, sorry about that. I didn't realise that this was a dirty commit. I've split out these cleanup changes into #13139.